k9mail/src/main/res/values/strings.xml

```diff
@@ -347,6 +347,9 @@ Please submit bug reports, contribute new features and ask questions at
     <string name="global_settings_privacy_hide_useragent_detail">Remove K-9 User-Agent from mail headers</string>
     <string name="global_settings_privacy_hide_timezone">Hide timezone</string>
     <string name="global_settings_privacy_hide_timezone_detail">Use UTC instead of local timezone in mail headers and reply header</string>
+    <string name="global_settings_privacy_hide_hostname_when_connecting">Hide hostname</string>
+    <string name="global_settings_privacy_hide_hostname_when_connecting_detail">Use localhost instead of hostname/IP when
```

Comment:
The message should probably include that this is specific to SMTP. Also there's no guarantee this leads to the hostname not showing up in email headers. Maybe we should make the description more technical/accurate. It's probably better to leave some people wondering what this does rather than make them think the settings protects their privacy when that's not something we know for sure.

Maybe: `Identify as 'localhost' when connecting to SMTP servers`.